src/ocean/time/timeout/TimeoutManager.d

```diff
@@ -228,6 +227,37 @@ abstract class TimeoutManagerBase : ITimeoutManager
 
         /***********************************************************************
 
+            Same as `unregister` but also removes expirty from list of currently
+            expired registrations. That means it can be called from `timeout`
+            callbacks, contrary to `unregister`.
+
+            Params:
+                expiry = expiry token returned by register() when registering
+                         the IExpiryRegistration instance to unregister
+
+        ***********************************************************************/
+
+        protected void drop ( ref Expiry expiry )
+        {
+            // If this expiry is already present in the expired registration
+            // list that is processed right now, remove it. This makes possible
+            // to disable timer events from timeout callbacks of other timer
+            // events, even if they compete for the same time slot.
+
+            IExpiryRegistration registration =
+                *this.outer.expiry_to_client.get(&expiry);
+            foreach (ref pending; this.outer.expired_registrations[])
+            {
+                if (pending is registration)
```

Comment:
Could you add a comment on the `this.outer.expired_registrations` field stating that the elements may be set to null by this method?